lib/srv/app/azure/handler_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+)
+
+func TestForwarder_getToken(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name string
+
+		config HandlerConfig
+
+		managedIdentity string
+		scope           string
+
+		wantToken *azcore.AccessToken
+		checkErr  require.ErrorAssertionFunc
+	}{
+		{
+			name: "base case",
+			config: HandlerConfig{
+				getAccessToken: func(ctx context.Context, managedIdentity string, scope string) (*azcore.AccessToken, error) {
+					if managedIdentity != "MY_IDENTITY" {
+						return nil, trace.BadParameter("wrong managedIdentity")
+					}
+					if scope != "MY_SCOPE" {
+						return nil, trace.BadParameter("wrong scope")
+					}
+					return &azcore.AccessToken{Token: "foobar"}, nil
+				},
+			},
+			managedIdentity: "MY_IDENTITY",
+			scope:           "MY_SCOPE",
+			wantToken:       &azcore.AccessToken{Token: "foobar"},
+			checkErr:        require.NoError,
+		},
+		{
+			name: "timeout",
+			config: HandlerConfig{
+				getAccessToken: func(ctx context.Context, managedIdentity string, scope string) (*azcore.AccessToken, error) {
+					time.Sleep(getTokenTimeout * 2)
```

Comment:
Don't sleep in test -   adding 2 seconds delay in unit test in not good idea.

Reply:
`getToken` uses `context.WithTimeout`, which isn't easily testable. I could potentially use https://pkg.go.dev/github.com/jonboulle/clockwork, but I'd have to rewrite it to use `clock.After`. Perhaps I will.

Reply:
Done: `clock.After` is used, we no longer have `time.Sleep` in tests.

Reply:
Actually, I have only fixed the GCP implementation; the Azure implementation needs a fix too.

Reply:
Done for Azure too.